lib/src/settings.rs

```diff
@@ -71,6 +82,34 @@ impl UserSettings {
         Ok(RepoSettings { _config: config })
     }
 
+    fn create_new_rng(&self) -> JJRng {
+        match self
+            .config
+            .get_string("debug.randomness-seed")
+            .ok()
+            .and_then(|str| str.parse().ok())
+        {
+            Some(seed) => JJRng::seed_from_u64(seed),
+            None => JJRng::from_entropy(),
+        }
+    }
+
+    /// Wraps Rng::gen but only requires an immutable reference to UserSettings.
+    ///
+    /// Note that the first time this is called, the random number generator
+    /// seed will be fixed regardless of further changes to the
+    /// `debug.randomness_seed` setting.
+    pub fn random_gen<T>(&self) -> T
```

Comment:
nit: `random_gen()` made me think it would return a random-number generator. Maybe `gen_random()` instead?